include/circt/Dialect/FIRRTL/FIRRTLStatements.td

```diff
@@ -50,12 +50,13 @@ def ConnectOp : FIRRTLOp<"connect", [FConnectLike]> {
   let hasCanonicalizer = true;
 }
 
-def StrictConnectOp : FIRRTLOp<"strictconnect", [SameTypeOperands, FConnectLike]> {
```

Comment:
Can you make a SameAnonTypeOperands constraint?

Reply:
Done, thank you for the suggesion!